### ISSUE TITLE
MongoDB exporter showing authentication failed

### ISSUE THREAD

**Issue by pankaj-dahiya** [bitnami/charts#7592]
time="2021-09-22T21:05:58Z" level=error msg="Problem gathering the mongo server version: connection() : auth error: sasl conversation error: unable to authenticate using mechanism \"SCRAM-SHA-256\": (AuthenticationFailed) Authentication failed." source="mongodb_collector.go:196"

**Describe the bug**
Enabling MongoDB metrics will not really enable the metrics as it is giving authentication failed errors.

**To Reproduce**
Steps to reproduce the behaviour:

Just enable metrics via metrics.enabled:true

**Expected behaviour**
Enable metrics and boom all the metrics will be shown in Prometheus. But only 5 basic metrics are showing in prometheous.

**Comment by migruiz4** [comment]
Hi @pankaj-dahiya,

Thank you for reporting this issue.

This bug was a bit tricky to find, but I was able to fix it using a workaround.

What is happening is that the `mongodb_exporter` starts simultaneously with the mongodb container and tries to connect before the root user has been created, causing it to fail.

The mongodb exporter apparently doesn't try to reconnect after the authentication failure, exposing just its internal metrics.

The workaround is quite simple, just adding a sleep as it follows should fix the problem.
```
args: 
  - |
  sleep 10; /bin/mongodb_exporter --web.listen-address ":{{ .Values.metrics.containerPort }}" --mongodb.uri "{{ include "mongodb.mongodb_exporter.uri" . }}" {{ .Values.metrics.extraFlags }}

```
But I'm looking into a proper fix by modifying the MongoDB readiness so it does not show as ready before the root/metrics user has been created (when authentication is enabled), and see if it fixes the issue with the exporter.

**Comment by nc-kfl** [comment]
Is this something which is planned to fix fixed in a future version?

**Comment by javsalgar** [comment]
Hi,

I think we should work on adding a small step just like @migruiz4 mentioned. @migruiz4 were you able to fix the readiness issue?

**Comment by juan131** [comment]
Let's keep this open until @migruiz4 responds about the changes in the probes he was working on.

**Comment by migruiz4** [comment]
Hi,

Sorry for the late response. I have submitted this PR with my suggested fix #8537
